asapdiscovery-genetics/asapdiscovery/genetics/seq_alignment.py

```diff
@@ -337,59 +443,164 @@ def do_MSA(
-        "E": "cyan",  # Glutamic acid
-        "G": "magenta",  # Glycine
-        "H": "pink",  # Histidine
-        "I": "brown",  # Isoleucine
-        "L": "gray",  # Leucine
-        "K": "lime",  # Lysine
-        "M": "teal",  # Methionine
-        "F": "navy",  # Phenylalanine
-        "P": "olive",  # Proline
-        "S": "maroon",  # Serine
-        "T": "silver",  # Threonine
-        "W": "gold",  # Tryptophan
-        "Y": "skyblue",  # Tyrosine
-        "V": "violet",  # Valine
-        "-": "white",
-    }
-    colors = [aa_colors[i] for i in text]
+    colors = []
+    for aa in text:
+        amino = AAcid(aa)
+        colors.append(amino.get_aminoacid_color())
     return colors
+
+
+# Defining colors for each protein residue
+def get_colors_by_aa_group(seq: str, max_missmatch=2):
+    """Make fill and text color for exact and group aminoacid matches
+
+    Parameters
+    ----------
+    seq : str
+        String with protein sequence
+    max_missmatch : int, optional
+       Maximum number of group missmatches after which match won't be highlighted, by default 2
+
+    Returns
+    -------
+    str, list[str]
+        Fill color, Font colors
+    """
+
+    from collections import Counter
+
+    seq_len = len(seq)
+    aa_groups = [AAcid(aa).get_aminoacid_group() for aa in seq]
+    aa_counts = Counter(aa_groups)
+    max_group, max_group_count = aa_counts.most_common(1)[0]
+    # Default font color
+    font_color = ["black"] * seq_len
+    # Check the case where all aa's are the same
+    if seq == seq_len * seq[0]:
+        color = "red"
+    # Check the case where all aa's belong to the same group (with some max missmatches)
+    elif max_group_count >= seq_len - max_missmatch:
+        if max_group is None:  # In case most "matches" are gaps
+            color = "white"
+        else:
+            color = "yellow"
+            # Make font red for missmatches
+            font_color = ["black" if item == max_group else "red" for item in aa_groups]
+    else:
+        color = "white"
+    return color, font_color
+
+
+class AAcid:
+    def __init__(self, letter_id):
+        """An aminoacid object
+
+        Parameters
+        ----------
+        letter_id : str
+            Amino Acid one or three-letter identifier
+        """
+        from MDAnalysis.lib.util import convert_aa_code
+
+        # An empty aminoacid
+        if letter_id == "-":
+            self.one_letter_id = letter_id
+            self.three_letter_id = None
+        elif len(letter_id) == 1:
+            self.one_letter_id = letter_id
+            self.three_letter_id = convert_aa_code(letter_id)
+        elif len(letter_id == 3):
+            self.three_letter_id = letter_id
+            self.one_letter_id = convert_aa_code(letter_id)
+        else:
+            raise ValueError(
+                "The input must be either the aminoacid 1 or 3-letter code"
+            )
+
+    def get_aminoacid_group(self):
+        agroups = {
```

Comment:
As this is only needed once per interpreter you can make it a constant outside of class scope

```python 

_AMINO_ACID_GROUPS = {...}

class AminoAcid:

    
    def __init__(...):
    
   def get_aminoacid_group(self):
       for key in _AMINO_ACID_GROUPS:
           ...
```